source/Mlos.Python/mlos/Optimizers/BayesianOptimizer.py

```diff
@@ -102,19 +102,27 @@ def get_all_observations(self):
         return self._feature_values_df.copy(), self._target_values_df.copy()
 
     @trace()
-    def suggest(self, random=False, context=None):
+    def suggest(self, random=False, context: Point = None):
         # TODO: pass context to the suggest method
+        if self.optimization_problem.context_space is not None:
+            if context is None:
+                raise ValueError("Non-empty context space but no context given.")
+            assert context in self.optimization_problem.context_space
         random = random or self.num_observed_samples < self.optimizer_config.min_samples_required_for_guided_design_of_experiments
-        suggested_config = self.experiment_designer.suggest(random=random)
+        suggested_config = self.experiment_designer.suggest(random=random, context_values_dataframe=context.to_dataframe())
         assert suggested_config in self.optimization_problem.parameter_space
         return suggested_config
 
     @trace()
-    def register(self, feature_values_pandas_frame, target_values_pandas_frame):
+    def register(self, config_values_pandas_frame, target_values_pandas_frame, context_values_pandas_frame=None):

```

Comment:
Could we name it parameter_values_df or something similar? I mainly mean the rename from config to parameters to keep it in line with the name of the parameter space. The same would apply to the variable names below.

Reply:
should I get rid of the ``pandas_frame`` everywhere if I'm doing the rename?